arrows/klv/tests/test_klv_read_write.cxx

```diff
@@ -0,0 +1,437 @@
+  KLV_WRITE_INT( uint32_t, 4, std::numeric_limits< uint32_t >::max() )
+  KLV_WRITE_INT( uint64_t, 8, std::numeric_limits< uint64_t >::max() )
+  KLV_WRITE_INT( int8_t,   1, std::numeric_limits< int8_t   >::max() )
+  KLV_WRITE_INT( int16_t,  2, std::numeric_limits< int16_t  >::max() )
+  KLV_WRITE_INT( int32_t,  4, std::numeric_limits< int32_t  >::max() )
+  KLV_WRITE_INT( int64_t,  8, std::numeric_limits< int64_t  >::max() )
+
+  // Unsigned - smaller than native size
+  KLV_WRITE_INT( uint16_t, 1, 0x00 )
+  KLV_WRITE_INT( uint32_t, 3, 0x001122 )
+  KLV_WRITE_INT( uint64_t, 5, 0x0011223344 )
+
+  // Signed, positive - smaller than native size
+  KLV_WRITE_INT( int16_t,  1, 0x00 )
+  KLV_WRITE_INT( int32_t,  3, 0x001122 )
+  KLV_WRITE_INT( int64_t,  5, 0x0011223344 )
+
+  // Signed, negative - smaller than native size
+  KLV_WRITE_INT( int16_t,  1, -0x11 )
+  KLV_WRITE_INT( int32_t,  3, -0x112233 )
+  KLV_WRITE_INT( int64_t,  5, -0x1122334455 )
+
+  // Unsigned - too few bytes allowed
+  KLV_WRITE_INT_ERR( uint16_t, 1, 1ull << ( 1 * 8 ),
+                     kv::metadata_type_overflow )
+  KLV_WRITE_INT_ERR( uint32_t, 3, 1ull << ( 3 * 8 ),
+                     kv::metadata_type_overflow )
+  KLV_WRITE_INT_ERR( uint64_t, 5, 1ull << ( 5 * 8 ),
+                     kv::metadata_type_overflow )
+
+  // Signed, positive - too few bytes allowed
+  KLV_WRITE_INT_ERR( int16_t,  1, 1ull << ( 0 * 8 + 7 ),
+                     kv::metadata_type_overflow )
+  KLV_WRITE_INT_ERR( int32_t,  3, 1ull << ( 2 * 8 + 7 ),
+                     kv::metadata_type_overflow )
+  KLV_WRITE_INT_ERR( int64_t,  5, 1ull << ( 4 * 8 + 7 ),
+                     kv::metadata_type_overflow )
+
+  // Signed, negative - too few bytes allowed
+  KLV_WRITE_INT_ERR( int16_t,  1, -129,          kv::metadata_type_overflow )
+  KLV_WRITE_INT_ERR( int32_t,  3, -8388609,      kv::metadata_type_overflow )
+  KLV_WRITE_INT_ERR( int64_t,  5, -549755813889, kv::metadata_type_overflow )
+
+#undef KLV_ROUND_TRIP
+#undef KLV_WRITE_INT_ERR
+}
+
+// ----------------------------------------------------------------------------
+TEST ( klv, read_ber )
+{
+#define KLV_READ_BER( T, value, ... )                         \
+  {                                                           \
+    vec_t const data = { __VA_ARGS__ };                       \
+    auto it = data.begin();                                   \
+    EXPECT_EQ( value, klv_read_ber< T >( it, data.size() ) ); \
+    EXPECT_EQ( data.end(), it );                              \
+  }
+
+#define KLV_READ_BER_ERR( T, length, e, ... )                 \
+  {                                                           \
+    vec_t const data = { __VA_ARGS__ };                       \
+    auto it = data.begin();                                   \
+    EXPECT_THROW( klv_read_ber< T >( it, length ), e );       \
+    EXPECT_EQ( data.begin(), it );                            \
+  }
+
+  // Short form
+  KLV_READ_BER( uint8_t, 0,
```

Comment:
Odd formatting

Reply:
This looks like a case of Uncrustify confused by the lack of a `;`? Maybe consider rewriting the macros wrapped with `do ... while(0)`.